configure.ac

```diff
@@ -198,23 +198,57 @@ AC_ARG_WITH([cuse],
             [],
             [with_cuse=check]
 )
+AC_ARG_WITH([fuse3],
+            AS_HELP_STRING([--with-fuse3],[build with libfuse3 instead of libfuse2]),
+            [],
+            [with_fuse3=check]
+)
 
 if test "$with_cuse" != "no"; then
-    LIBFUSE_CFLAGS=$(pkg-config fuse --cflags 2>/dev/null)
-    if test $? -ne 0; then
```

Comment:
It was checking whether fuse was installed and if it could not be found but the user requested it, it would throw the error. Otherwise it would build with cuse just if fuse was found. I think this behavior should remain, but now first test for fuse3 and if not available then fall back to fuse.

Another good thing would be to display the LIBFUSE_CFLAGS and LIBFUSE_LIBS at the bottom of the script so the user sees what it links against.